pmda/rdf.py

```diff
@@ -0,0 +1,158 @@
+
+import numpy as np
+
+from MDAnalysis.lib.distances import distance_array
+from MDAnalysis.lib.util import blocks_of
+
+from .parallel import ParallelAnalysisBase
+
+
+class InterRDF(ParallelAnalysisBase):
+    """Intermolecular pair distribution function
+
+    InterRDF(g1, g2, nbins=75, range=(0.0, 15.0))
+
+    Arguments
+    ---------
+    g1 : AtomGroup
+      First AtomGroup
+    g2 : AtomGroup
+      Second AtomGroup
+    nbins : int (optional)
+          Number of bins in the histogram [75]
+    range : tuple or list (optional)
+          The size of the RDF [0.0, 15.0]
+    exclusion_block : tuple (optional)
+          A tuple representing the tile to exclude from the distance
+          array. [None]
+    start : int (optional)
+          The frame to start at (default is first)
+    stop : int (optional)
+          The frame to end at (default is last)
+    step : int (optional)
+          The step size through the trajectory in frames (default is
+          every frame)
+
+    Example
+    -------
+    First create the :class:`InterRDF` object, by supplying two
+    AtomGroups then use the :meth:`run` method ::
+
+      rdf = InterRDF(ag1, ag2)
+      rdf.run()
+
+    Results are available through the :attr:`bins` and :attr:`rdf`
+    attributes::
+
+      plt.plot(rdf.bins, rdf.rdf)
+
+    The `exclusion_block` keyword allows the masking of pairs from
+    within the same molecule.  For example, if there are 7 of each
+    atom in each molecule, the exclusion mask `(7, 7)` can be used.
+
+    .. versionadded:: 0.2.0
+
+    """
+    # pylint: disable=redefined-builtin
+    # continue to use 'range' as long as MDAnalysis uses it so that
+    # the user interface remains consistent
+    def __init__(self, g1, g2,
+                 nbins=75, range=(0.0, 15.0), exclusion_block=None):
+        u = g1.universe
+        super(InterRDF, self).__init__(u, (g1, g2))
+
+        # collect all atomgroups with the same trajectory object as universe
+        trajectory = u.trajectory
+        self.nA = len(g1)
+        self.nB = len(g2)
+        self.nf = trajectory.n_frames
+        self.rdf_settings = {'bins': nbins,
+                             'range': range}
+        self._exclusion_block = exclusion_block
+
+        edges = np.histogram([-1], **self.rdf_settings)[1]
+        self.edges = edges
+        self.bins = 0.5 * (edges[:-1] + edges[1:])
+
+    # pylint: enable=redefined-builtin
+
+    def _prepare(self):
+        # Empty histogram to store the RDF
+        self.count = self.bins * 0.0
+
+        # Need to know average volume
+        self.volume = 0.0
+
+    def _single_frame(self, ts, atomgroups):
+        g1, g2 = atomgroups
+        u = g1.universe
+        d = distance_array(g1.positions, g2.positions,
+                           box=u.dimensions)
+        # If provided exclusions, create a mask of _result which
+        # lets us take these out
+        self._exclusion_mask = None
```

Comment:
This code must be changed in such a way that `self._exclusion_mask` and `self._maxrange` are *never written to in `_single_frame()`.